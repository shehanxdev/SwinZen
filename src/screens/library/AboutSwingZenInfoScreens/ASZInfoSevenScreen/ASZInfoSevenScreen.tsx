import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { AboutSZInfoSevenPoints, Color, TextAlignment, TextVariant } from '@sz/constants';

import { BaseLibraryScreen } from '../../components';

const TEST_ID_PREFIX = 'ASZInfoSevenScreenTestID';

export function ASZInfoSevenScreen() {
  return (
    <BaseLibraryScreen>
      <View style={tw`mx-4 mt-8`} testID={TEST_ID_PREFIX}>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
          {`Note this is available only with paid subscription.\n`}
        </Text>
        {AboutSZInfoSevenPoints.map((data, index) => (
          <View key={`${TEST_ID_PREFIX}-Point-${index}`}>
            <Text color={Color.Neutral.White} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
              {data.title}
            </Text>
            <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
              {data.body}
              {`\n`}
            </Text>
          </View>
        ))}
      </View>
    </BaseLibraryScreen>
  );
}
