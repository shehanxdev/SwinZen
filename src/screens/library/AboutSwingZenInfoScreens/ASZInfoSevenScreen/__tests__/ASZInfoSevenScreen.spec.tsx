import { render } from '@testing-library/react-native';
import * as React from 'react';

import { ASZInfoSevenScreen } from '../ASZInfoSevenScreen';

describe('ASZInfoSeven Screen', () => {
  const testID = 'ASZInfoSevenScreenTestID';

  const getRenderedScreen = () => render(<ASZInfoSevenScreen />);

  it(`should render ASZInfoSeven Screen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the ASZInfoSeven Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
