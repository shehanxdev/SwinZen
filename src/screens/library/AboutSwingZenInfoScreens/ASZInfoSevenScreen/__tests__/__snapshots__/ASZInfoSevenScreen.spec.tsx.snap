// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASZInfoSeven Screen should render ASZInfoSeven Screen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279917603,
      4278984207,
      4278650631,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/images/grassBackground.jpg",
      }
    }
    style={
      {
        "opacity": 0.1,
        "position": "absolute",
      }
    }
  />
  <RCTSafeAreaView
    style={
      {
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
      >
        <View>
          <View
            style={
              {
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 32,
              }
            }
            testID="ASZInfoSevenScreenTestID"
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Note this is available only with paid subscription.

            </Text>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Setup/Hips Centered:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Determines if hips are properly positioned in setup between your feet positions at address.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Setup/Spine Tilt:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Measures proper spine tilt away from the target based on driver or iron shot at address.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Setup/Shoulder Tilt:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Measures proper shoulder tilt away from target at address position.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Takeaway:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Determines if the club and arms are taken back in one piece during initial movement in backswing.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Lead Arm:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks the angle of lead arm bend during the backswing to determine if excessive bend.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Hip Sway:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks position of hips thru backswing to determine proper and efficient movement.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Vertical Head Movement:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks position of head thru backswing to determine proper vertical movement.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Lateral Head Movement:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks position of head thru backswing to determine proper lateral movement.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Shoulder Tilt:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks shoulder tilt thru backswing to determine proper and efficient movement.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Full Backswing:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks clubhead to determine if too far rotation in backswing.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Downswing/Club Lag Release Point:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks the club lag angle to determine when the angle is released at the most efficient point for power and consistency.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Downswing/Vertical Head Movement:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks position of head thru downswing to determine proper vertical movement.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Downswing/Lateral Head Movement:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks position of head thru downswing to determine proper lateral movement.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Downswing/Weight Transfer:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks the body movement thru impact to determine proper and efficient movement.
                

              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
